src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorFormattingService.cs

```diff
@@ -171,7 +171,9 @@ public async Task<TextEdit[]> FormatSnippetAsync(DocumentContext documentContext
         if (collapseEdits)
         {
             var collapsedEdit = MergeEdits(edits, originalText);
-            if (collapsedEdit.NewText.Length == 0)
+            if (collapsedEdit.NewText.Length == 0 &&
+                collapsedEdit.Range.Start.Line == collapsedEdit.Range.End.Line &&
+                collapsedEdit.Range.Start.Character == collapsedEdit.Range.End.Character)
```

Comment:
This change is required because otherwise the edit to remove the `@` on `@"goo"` was being dropped. I've honestly no idea why we would have been filtering out empty edits, and can only guess that the intent was to filter out useless edits. No existing formatting tests break with this change, so we've got that going for us at least.